### ISSUE TITLE
This GPT is cannot even perform a simple analysis

### ISSUE THREAD

**Issue by Flydawgs** [bloodlinealpha/NHL-Stats-and-Analysis-Expert#1]
Kudos for making this GPT but it's worthless if it can't even execute a simple analysis. I tried over 10 times to have it tell me which players in the NHL have the most 5+ shots-on-goal games this season - And it spat out gibberish. I narrow the ask to show me how many times did David Pastrnak have 5+ SOG to date this season, and the number was wrong by half.

**Comment by bloodlinealpha** [comment]
Thanks for letting me know. I did make this GPT as an example, with full source code so others could improve upon it. I will slowly chip away at making it more versatile, but currently it is only capable of relatively simple analysis. I know with the GPT store now people are using it more and I agree it would be great if it worked better.

The main issue being the GPT is unable to handle large amount of data coming back from the NHL API. I will look into some ways to fine tune the API calls so it only brings back what is necessary. The NHL API is a bit of a mess as it recently changed and is mostly undocumented.

As for some tips I would say to try and be as specific as possible. As you can see below it is trying its best with the API params given in the GPT Actions, but there is no query parameter for Shots on goal > #.

![image](https://github.com/bloodlinealpha/NHL-Stats-and-Analysis-Expert/assets/22661326/870b1920-e316-4b59-bd05-fccf2e28cbb3)

As for individual players stats , game by game, I will look into adding that as it currently has just grabs the data from the players page (which is a large json) and tries to figure something out.

Again feel free to use the code in the repo, recreate the GPT, and add and change things if you have a specific use case.

I hope that helps!

**Comment by bloodlinealpha** [comment]
@Flydawgs. The commit above adds functionality for game logs by player. I also updated the GPT Actions with this functionality, so it is live. It is still not perfect for trying to find stats game by game across the whole season as the data returned can be very large. I have created a new issue (#2) to track and work on this. Feel free to add any idea's to this issue.

When asking about individual games for a player it works great!


![image](https://github.com/bloodlinealpha/NHL-Stats-and-Analysis-Expert/assets/22661326/719ff263-c173-43b2-ad7e-7a3f81c1eb75)
